netty/src/test/java/io/grpc/netty/NettyAdaptiveCumulatorTest.java

```diff
@@ -582,6 +583,7 @@ public CompositeByteBuf addFlattenedComponents(boolean increaseWriterIndex,
         assertEquals(0, compositeRo.numComponents());
       } finally {
         compositeRo.release();
+        in.release();
```

Comment:
I think we should keep it above`fail("Cumulator didn't throw");`

Reply:
We need to release when there is an error since the new paradigm is that only successes change ownership.

Reply:
Discussed IRL, we do want to move it, but to another place